core/example/tutorial/14_unmanaged_view/parallel_for_example.cpp

```diff
@@ -0,0 +1,137 @@
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the Cabana library. Cabana is distributed under a   *
+ * BSD 3-clause license. For the licensing terms see the LICENSE file in    *
+ * the top-level directory.                                                 *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+
+#include <Cabana_Core.hpp>
+#include <Cabana_Sort.hpp>
+
+#include <cassert>
+
+//---------------------------------------------------------------------------//
+// parallel for using an unmanaged view.
+//---------------------------------------------------------------------------//
+void parallelForExample()
+{
+    /* Declare general run parameters */
+    using MemorySpace = Kokkos::HostSpace;
+    using ExecutionSpace = Kokkos::DefaultHostExecutionSpace;
+    using DeviceType = Kokkos::Device<ExecutionSpace,MemorySpace>;
+    const int VectorLength = 8;
+
+    /*
+     * Declare Underlying data type
+    */
+
+    // Regular class form
+    class Data {
+        public:
+            double a[VectorLength];
+            double b[VectorLength];
+    };
+
+    // Cabana MemberTypes form:
+    using DataTypes = Cabana::MemberTypes<double,double>;
+
+
+    /*
+     * Create the AoSoA.
+    */
+    const int num_tuple = 128;
+    const int num_soa = 128/8; // Be careful with the int division..
+
+    Data* local_data = new Data[num_tuple];
+
+    // This is equivalent to a Cabana AoAoS, which looks like:
+    //    Cabana::AoSoA<DataTypes,DeviceType,VectorLength> aosoa( "my_aosoa", num_tuple );
+
+    /*
+     * Populate user data.
+    */
+    for (int i = 0; i < num_soa; i++)
+    {
+        for (int j = 0; j < VectorLength; j++)
+        {
+            local_data[i].a[j] = i;
+            local_data[i].b[j] = j;
+        }
+    }
+
+    /*
+    * Convert the user allocated memory into an unmanaged AoSoA
+    */
+
+    // Define the underlying AoSoA type to hold the unmanaged data
+    using AoSoA_t = Cabana::AoSoA<
+        DataTypes,
+        MemorySpace,
+        VectorLength,
+        Kokkos::MemoryUnmanaged
+    >;
+
+    // Do convert
+    auto ptr = reinterpret_cast<typename AoSoA_t::soa_type*>(local_data);
+    AoSoA_t aosoa( ptr, num_soa, num_tuple );
+
+    /*
+    * Inspect Data In the Unmanaged AoSoA
+    */
+    auto slice_a = aosoa.slice<0>(); // a = 0
+    auto slice_b = aosoa.slice<1>(); // b = 1
+
+    // Look at the data in the AosoA
+    for (int i = 0; i < num_tuple; i++)
+    {
+        if ( slice_a(i) != i / VectorLength )
+        {
+            // Unexpected Value
+            printf("%d: Unexpected %e != %e \n", __LINE__, slice_a(i), i / VectorLength);
```

Comment:
You want to go ahead and assert it?  (Since you included the header `<cassert>` and don't actually use it)

Reply:
Yeah I would `assert` or `throw` here

Reply:
Well maybe not actually - this seems to be more like a check in a unit test - would we want to halt the program here?

Reply:
Yea thats exactly the cycle I went through. I wrote it as asserts, but then ran it in Release mode and it didn't tell me when I screwed up, so decided that a manual `if` was more robust

Reply:
I don't have a strong opinion.  We have debug builds in the CI so we would catch bugs.  If you decide not to have assertions then remove `#include <cassert>`.

Reply:
Let's stick with `printf`. I cut them down a bit, now they're a bit less offensive too